app/controllers/api/custom_buttons_controller.rb

```diff
@@ -46,10 +44,7 @@ def create_or_update_resource_action(data)
       return nil if data.empty?
       id = data["id"]
       if id.present?
-        resource_action = ResourceAction.find(id)
-        data.except!("id")
-        resource_action.update!(data.deep_symbolize_keys)
-        resource_action
+        data.except!("id").each_with_object(ResourceAction.find(id)) { |(k, v), ra| ra.send("#{k}=", v) }
```

Comment:
Why `each_with_object` instead of the `update!` that was here before?

Reply:
Also, this is no longer saving any changes, right?

Reply:
doesn't need to, as Keenan pointed out, the update here isn't necessary since it's handled by line 24

Reply:
Line 24 is updating the CustomButton, not the ResourceAction

Reply:
the button that gets saved with the resource action returned on 22, the update isn't needed twice

Reply:
@d-m-u can you point us to where the update happens?

It is a little tricky to see how this code reaches line 24

Reply:
Don't think this works.